sonar-go-checks/src/main/java/org/sonar/go/checks/ElseIfWithoutElseCheck.java

```diff
@@ -76,20 +81,42 @@ private static boolean isTopLevelIf(CheckContext ctx, IfTree ifTree) {
     return true;
   }
 
-  private static boolean endsWithReturnBreakOrThrow(IfTree ifTree) {
+  private static boolean endsWithReturnBreakOrPanic(IfTree ifTree) {
     Tree thenBranch = ifTree.thenBranch();
     if (thenBranch instanceof BlockTree blockTree) {
-      List<Tree> statements = blockTree.statementOrExpressions();
+      List<Tree> statements = blockTree.statementOrExpressions()
+        .stream().filter(TreeUtils.IS_NOT_SEMICOLON)
+        .toList();
       if (!statements.isEmpty()) {
         Tree lastStmt = statements.get(statements.size() - 1);
-        return isReturnBreakOrThrow(lastStmt);
+        return isReturnBreakOrPanic(lastStmt);
       }
     }
     // Curly braces can be omitted when there is only one statement inside the "if"
-    return isReturnBreakOrThrow(thenBranch);
+    return isReturnBreakOrPanic(thenBranch);
   }
 
-  private static boolean isReturnBreakOrThrow(Tree tree) {
-    return tree instanceof JumpTree || tree instanceof ReturnTree || tree instanceof ThrowTree;
+  private static boolean isReturnBreakOrPanic(Tree tree) {
+    return tree instanceof JumpTree || tree instanceof ReturnTree || isPanicCall(tree);
+  }
+
+  private static boolean isPanicCall(Tree tree) {
```

Comment:
The method doesn't work if `painc()` is not the last call in branch, e.g.:
```
        if x == 0 {
            break;
        } else if x == 1 {
            return;
        } else if x == 3 {
            panic(1);
            doSomething();
        }
```

Please add the example to unit tests

Reply:
I will also add similar tests for `break` and `return`, as even code placed after `panic()` are reported as unreachable code by go compiler.